internal/framework/controller/predicate/gatewayclass_test.go

```diff
@@ -31,4 +32,6 @@ func TestGatewayClassPredicate(t *testing.T) {
 	g.Expect(p.Update(event.UpdateEvent{ObjectOld: gc, ObjectNew: gc2})).To(BeTrue())
 	g.Expect(p.Update(event.UpdateEvent{ObjectOld: gc2, ObjectNew: gc})).To(BeTrue())
 	g.Expect(p.Update(event.UpdateEvent{ObjectOld: gc2, ObjectNew: gc2})).To(BeFalse())
+	g.Expect(p.Delete(event.DeleteEvent{Object: nil})).To(BeFalse())
+	g.Expect(p.Delete(event.DeleteEvent{Object: gc2})).To(BeFalse())
```

Comment:
Missing one test case where the type of the Object isn't GatewayClass. I think this should do it:
```
g.Expect(p.Delete(event.DeleteEvent{Object: &v1.HTTPRoute{}})).To(BeFalse())
```

Reply:
Yeah will address soon.

Reply:
addressed